sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/IntegrationTestBase.java

```diff
@@ -204,46 +173,65 @@ public ProxyOptions getProxyConfiguration() {
     }
 
     /**
-     * Creates a new instance of {@link ServiceBusClientBuilder} with the default integration test settings and uses a
-     * connection string to authenticate.
+     * Creates a new instance of {@link ServiceBusClientBuilder} with authentication set up in {@link TestMode#LIVE} and
+     * {@link TestMode#RECORD} modes.
+     *
+     * @return the builder with authentication set up.
+     * @throws org.opentest4j.TestAbortedException if the test mode is {@link TestMode#PLAYBACK}.
      */
-    protected ServiceBusClientBuilder getBuilder() {
-        return getBuilder(false);
+    protected ServiceBusClientBuilder getAuthenticatedBuilder() {
+        final TestMode mode = super.getTestMode();
+        final ServiceBusClientBuilder builder = new ServiceBusClientBuilder();
+        if (mode == TestMode.LIVE) {
+            final String fullyQualifiedDomainName = TestUtils.getLiveFullyQualifiedDomainName();
+            assumeTrue(!CoreUtils.isNullOrEmpty(fullyQualifiedDomainName), "FullyQualifiedDomainName is not set.");
+            final TokenCredential credential = TestUtils.getPipelineCredential(credentialCached);
+            return builder.credential(fullyQualifiedDomainName, credential);
+        } else if (mode == TestMode.RECORD) {
+            final String connectionString = TestUtils.getConnectionString(false);
+            if (CoreUtils.isNullOrEmpty(connectionString)) {
+                final String fullyQualifiedDomainName = TestUtils.getLiveFullyQualifiedDomainName();
+                assumeTrue(!CoreUtils.isNullOrEmpty(fullyQualifiedDomainName), "FullyQualifiedDomainName is not set.");
+                final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+                return builder.credential(fullyQualifiedDomainName, credential);
+            } else {
+                return builder.connectionString(connectionString);
```

Comment:
On local sdk-developer setup, it is expected to use RECORD mode if running tests against real service is desired (Note: the AMQP traffic never gets recorded to files).

For local runs, the DefaultAzureCredential is enabled per recommendation. But I kept the connection string as an option for sdk-dev inner loop . If connection string is not set, then DefaultAzureCredential is attempted.

If/when we drop connection string local sdk-dev experience, it's simply about removing the above if condition. This is a follow up item not scoped to this pr.